### DISCUSSION TITLE
React Widget for a file uri

### DISCUSSION THREAD

**Discussion by vibsid0986** [eclipse-theia/theia#13680] [General] [answered]
Greetings Team,

I'm developing a React Widget to automatically display the application whenever a file with the extension .ext is accessed. Utilizing the WidgetOpenHandler, I've managed to implement this functionality. However, a crucial challenge has surfaced: there's a lack of synchronization between the application and individual instances of the file resource. In simpler terms, if multiple files with the .ext extension are opened, it's unclear which application corresponds to which file instance.

Could you please advise on the necessary bindings within my frontend module or any example to refer for? I aim to ensure seamless management of title, label, and state persistence, even during reloads, whenever a .ext file is opened or closed.

Your assistance on this matter would be immensely valuable.

Thank you in advance.

**Comment by msujew** [comment, marked as answer]
Hey @vibsid0986,

widgets that represent a resource should implement the `Navigatable` interface. It is used for Theia to determine which widget belongs to a given file. See also how the [notebook widget](https://github.com/eclipse-theia/theia/blob/b20a7513502fada4045b375f068152a7235e6c05/packages/notebook/src/browser/notebook-editor-widget.tsx#L72) for example is implemented.

**Comment by vibsid0986** [comment]
Thanks @msujew for your help, 
I was able to successfully register react widget corresponding to a file resource. Notebook widget example did help me a lot. :)
